src/main/java/org/apache/commons/cli/CommandLine.java

```diff
@@ -246,6 +331,44 @@ public String getOptionValue(char opt, String defaultValue)
     {
         return getOptionValue(String.valueOf(opt), defaultValue);
     }
+    
+    /**
+     * Retrieve the map of values associated to the option. This is convenient
+     * for options specifying Java properties like <tt>-Dparam1=value1
+     * -Dparam2=value2</tt>. The first argument of the option is the key, and
+     * the 2nd argument is the value. If the option has only one argument
+     * (<tt>-Dfoo</tt>) it is considered as a boolean flag and the value is
+     * <tt>"true"</tt>.
+     *
+     * @param option name of the option
+     * @return The Properties mapped by the option, never <tt>null</tt>
+     *         even if the option doesn't exists
+     * @since 1.4
+     */
+    public Properties getOptionProperties(Option option)
+    {
+        Properties props = new Properties();
+
+        for (Option processedOption : options)
+        {
+            if (processedOption.equals(option))
+            {
+                List<String> values = processedOption.getValuesList();
+                if (values.size() >= 2)
+                {
+                    // use the first 2 arguments as the key/value pair
+                    props.put(values.get(0), values.get(1));
+                }
+                else if (values.size() == 1)
+                {
+                    // no explicit value, handle it as a boolean
+                    props.put(values.get(0), "true");
+                }
+            }
+        }
+
+        return props;
```

Comment:
Why does it not belong to CLI271? Its an adoption of the previous String variant (what is now delegating to this method) that I adjusted a bit to remove the requirements of string matching.